index.js

```diff
@@ -1,39 +1,40 @@
 var EndpointPool,
-    _      = require('underscore'),
-    dns    = require('dns'),
-    Events = require('events'),
-    util   = require('util'),
+    _           = require('underscore'),
+    dns         = require('dns'),
+    Events      = require('events'),
+    PoolManager = require('./pool-manager'),
+    util        = require('util'),
 
-    DNS_LOOKUP_TIMEOUT = 1000,
-
-    CLOSED            = 0,  // closed circuit: endpoint is good to use
-    HALF_OPEN_READY   = 1,  // endpoint is in recovery state: offer it for use once
-    HALF_OPEN_PENDING = 2,  // endpoint recovery is in process
-    OPEN              = 3;  // open circuit: endpoint is no good
+    DNS_LOOKUP_TIMEOUT = 1000;
 
 /**
- * @param {String}   discoveryName The name of the service discovery host
- * @param {Number}   ttl           How long the endpoints are valid for. The service discovery endpoint will be checked on
- *                                 this interval.
- * @param {Number}   maxFailures   Number of failures allowed before the endpoint circuit breaker is tripped.
- * @param {Number}   failureWindow Size of the sliding window of time in which the failures are counted.
- * @param {Number}   resetTimeout  Amount of time before putting the circuit back into half open state.
- * @param {Function} onReady       Callback to execute when endpoints have been primed (updated for the first time)
+ * @param {String}    discoveryName       The name of the service discovery host
+ * @param {Number}    ttl                 How long the endpoints are valid for. The service discovery endpoint will be checked on
+ *                                        this interval.
+ * @param {{maxFailures: Number, failureWindow: Number, resetTimeout: Number}}
+ *                    ejectOnErrorConfig  How to handle endpoint errors. If specified, the following options must be defined:
+ *                                        - maxFailures: Number of failures allowed before the endpoint circuit breaker is tripped.
+ *                                        - failureWindow: Size of the sliding window of time in which the failures are counted.
+ *                                        - resetTimeout: Amount of time before putting the circuit back into half open state.
+ * @param {Function=} onReady       Callback to execute when endpoints have been primed (updated for the first time)
```

Comment:
@akovalev I did that just so you can differentiate b/w 'i dunno' and the other cases. In the README I point out it doesn't mean you actually have endpoints: https://github.com/soundcloud/dns-endpoint-pool/commit/586b2fc4478655b9c487d972bbb4a4efc6e170ac#diff-04c6e90faac2675aa89e2176d2eec7d8R54